### PR TITLE
[Improvement] Skip fetching Maven dependency snapshot update when building `dependencyList`

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- Skip Maven dependency snapshot update fetching to building `dependencyList` faster without waiting for fetching daily snapshots checking (only snapshots of submodules used).

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: build/dependency.sh [hunk 1]
```diff
@@ -81,6 +81,8 @@ cat >"${DEP_PR}"<<EOF
 
 EOF
 
+$MVN dependency:resolve
```
Comment (pan3793) [inline build/dependency.sh:84]: is it possible to combine it w/ line 35? just like 

```
$MVN dependency:resolve dependency:build-classpath -pl ...
```

Comment (bowenliang123) [reply build/dependency.sh:84]: Several reasons to do it separately,
1.  better not to do in one place as not sure whether the change will impact or contaminate the outputs for dependencyList.
2. solely resolving maven dependency and the output will be more clear to read and to know which process we are waiting for (eg. resolving deps , building_classpath)

**Comment by pan3793** [comment]
cc @cfmcgrady

**Comment by cfmcgrady** [comment]
> thus many daily updated snapshot releases will be downloaded again and this process is invisible and make it seems jammed with no console output.

do we have any daily updated snapshot dependencies?

**Comment by bowenliang123** [comment]
> > thus many daily updated snapshot releases will be downloaded again and this process is invisible and make it seems jammed with no console output.
> 
> do we have any daily updated snapshot dependencies?

Maven is redownloading the submodules from `apache.snapshots` every day after daily snapshots are published.
image

**Comment by cfmcgrady** [comment]
Since all dependencies are stable, and only project submodules are snapshot versions, why don't we skip snapshot updates when updating the dependencies list?

We can skip snapshot update by running

```
mvn goal --no-snapshot-updates
```
or
```
mvn goal -nsu
```

FYI: https://stackoverflow.com/questions/1348603/how-can-i-get-maven-to-stop-attempting-to-check-for-updates-for-artifacts-from-a

**Comment by cfmcgrady** [comment]
Since all dependencies are stable, and only project submodules are snapshot versions, why don't we skip snapshot updates when updating the dependencies list?

We can skip snapshot update by running

```
mvn goal --no-snapshot-updates
```
or
```
mvn goal -nsu
```

FYI: https://stackoverflow.com/questions/1348603/how-can-i-get-maven-to-stop-attempting-to-check-for-updates-for-artifacts-from-a

**Review by cfmcgrady — APPROVED** [review]

**Comment by bowenliang123** [comment]
> Since all dependencies are stable, and only project submodules are snapshot versions, why don't we skip snapshot updates when updating the dependencies list?
> 
> We can skip snapshot update by running
> 
> ```
> mvn goal --no-snapshot-updates
> ```
> 
> or
> 
> ```
> mvn goal -nsu
> ```
> 
> FYI: https://stackoverflow.com/questions/1348603/how-can-i-get-maven-to-stop-attempting-to-check-for-updates-for-artifacts-from-a

SGTM. Applied the suggestion of using `--no-snapshot-updates` option and also changed the title and description of this PR.

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**